src/main.c

```diff
@@ -3742,6 +3785,46 @@ static void create_per_thread_listeners(void)
     }
 }
 
+H2O_NORETURN static void *h2olog_thread(void *_ctx)
+{
+    while (1) {
+        int fd = accept(conf.h2olog.listening_fd, NULL, 0);
+        if (fd == -1) {
+            continue;
+        }
+        if (fcntl(fd, F_SETFD, FD_CLOEXEC) != 0) {
+            h2o_perror("failed to set FD_CLOEXEC");
+            close(fd);
+            continue;
+        }
+        if (fcntl(fd, F_SETFL, O_NONBLOCK) != 0) {
+            h2o_perror("failed to set O_NONBLOCK");
+            close(fd);
+            continue;
+        }
+        if (conf.h2olog.sndbuf != 0 && setsockopt(fd, SOL_SOCKET, SO_SNDBUF, &conf.h2olog.sndbuf, sizeof(conf.h2olog.sndbuf)) != 0) {
+            h2o_perror("failed to set SO_SNDBUF");
+            close(fd);
+            continue;
+        }
+
+        if (ptls_log_add_fd(fd) != 0) {
+            h2o_perror("failed to register fd to ptls_log");
```

Comment:
This code assumes that when `ptls_log_add_fd` fails, the cause of the error is set in `errno`.

I think that's an incorrect assumption. `ptls_log_add_fd` is a function that returns either 0 (on success) or an _error code_ when the operation fails. That's the general convention of functions defined by picotls.

Therefore, we should not be calling `h2o_perror`. Rather, we should either report the error code returned by `ptls_log_add_fd`, or simply say that the function has failed.

I tend to believe that the latter would be sufficient.